perfkitbenchmarker/benchmark_spec.py

```diff
@@ -247,7 +256,11 @@ def CreateVirtualMachine(self, opt_zone=None):
     if vm:
       return vm
 
-    vm_class = CLASSES[self.cloud][VIRTUAL_MACHINE][FLAGS.os_type]
+    vm_classes = CLASSES[self.cloud][VIRTUAL_MACHINE]
+    if FLAGS.os_type not in vm_classes:
+      raise errors.Error('The cloud "%s" does not support VMs of type "%s".' %
```

Comment:
I'd phrase it "VMs of type %type are currently not supported on provider %cloud"